### PR TITLE
Enforce the Content-Lenght for flares POST requests

### PR DESCRIPTION
**Pull request by CharlyF**
### What does this PR do?

Ensures we do not fail uploading flares because the Header is missing pieces required to validate the request for a flare.

### Motivation

As we introduced a new process to create flares that has a low memory footprint we are now using a `PipeReader` instead of a `Reader`. This is not supported by the net/http and as the body is not nil we need to enforce the content length to -1 as specified in the note here:
https://github.com/golang/go/blob/master/src/net/http/request.go#L269

0 does not work here because of this issue:
https://github.com/golang/go/issues/18117

### CODE REVIEW

File: releasenotes/notes/set-content-lenght-flare-3b220d6a30cef1b8.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Ensure flares with the new low memory footprint process can be uploaded.
```
Comment (mfpierre) [inline releasenotes/notes/set-content-lenght-flare-3b220d6a30cef1b8.yaml]: This should probably be moved to the dca release not as this wasn't released on the agent yet

**Review by clamoriniere — DISMISSED** [review]

File: pkg/flare/flare.go [hunk 2]
```diff
@@ -74,10 +74,14 @@ func SendFlareWithHostname(archivePath string, caseID string, email string, host
 
 	var url = mkURL(caseID)
 	request, err := http.NewRequest("POST", url, bodyReader)
-	request.Header.Set("Content-Type", writer.FormDataContentType())
 	if err != nil {
 		return "", err
 	}
+	request.Header.Set("Content-Type", writer.FormDataContentType())
```
Comment (albertvaka) [inline pkg/flare/flare.go]: Good catch.

**Review by albertvaka — DISMISSED** [review]

File: pkg/flare/flare.go [hunk 3]
```diff
@@ -38,8 +38,8 @@ func SendFlareWithHostname(archivePath string, caseID string, email string, host
 	//Write stuff to the pipe will block until it is read from the other end, so we don't load everything in memory
 	go func() {
 
-		defer bodyWriter.Close()
 		defer writer.Close()
+		defer bodyWriter.Close()
```
Comment (albertvaka) [inline pkg/flare/flare.go]: The defers execute in inverse order as they are written (they are "stacked"), so I think the previous order is the correct one (although probably doesn't matter much).

Comment (CharlyF) [reply pkg/flare/flare.go]: You are right, and it appears that it matters (unit tests broke when I changed).
Thanks!

**Review by albertvaka — APPROVED** [review]

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**